include/seqan3/argument_parser/detail/format_base.hpp

```diff
@@ -371,49 +352,59 @@ class format_help_base : public format_base
     }
 
     /*!\brief Adds a print_section call to parser_set_up_calls.
-     * \param[in] title The title of the section of the help page.
+     * \copydetails seqan3::argument_parser::add_section
      */
-    void add_section(std::string const & title)
+    void add_section(std::string const & title, option_spec const spec)
     {
-        parser_set_up_calls.push_back([this, title] ()
+        if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
         {
-            derived_t().print_section(title);
-        });
+            parser_set_up_calls.push_back([this, title] ()
```

Comment:
I take it that it cannot happen, that the object pointed to by this is destroyed before the lambda is invoked, since the vector is itself a member of the class, right?

Reply:
Yes exactly.